src/main/java/org/jitsi/jicofo/codec/JingleOfferFactory.java

```diff
@@ -434,51 +434,75 @@ private static void addAudioToContent(
             rtpDesc.addExtmap(ssrcAudioLevel);
         }
 
-        // a=rtpmap:111 opus/48000/2
-        PayloadTypePacketExtension opus = addPayloadTypeExtension(rtpDesc, 111, Constants.OPUS, 48000);
-        opus.setChannels(2);
+        if (config.opus.enabled())
+        {
+            // a=rtpmap:111 opus/48000/2
+            PayloadTypePacketExtension opus = addPayloadTypeExtension(rtpDesc, config.opus.pt(), Constants.OPUS, 48000);
+            // Opus is always signaled with 2 channels, regardless of 'stereo'
+            opus.setChannels(2);
+
+            // Though RED has a payload type of its own and can be used to encode multiple other payload types, we need
+            // it to be advertised with the same clock rate as opus, so its defined here.
+            if (config.opus.red.enabled())
+            {
+                PayloadTypePacketExtension red = addPayloadTypeExtension(rtpDesc, config.opus.red.pt(), "red", 48000);
+                red.setChannels(2);
+            }
```

Comment:
Do you want to set the fmtp for the red packet? (Note that the Jingle syntax for the red fmtp is poorly defined...)

Reply:
I don't think it is necessary, and it's currently not being used. Chrome does not set it in it's offer or answer (and strips it if it's set). It also exposes a bug in jitsi-videobridge (to make the XML format as undestood by lib-jitsi-meet work we need to use name=null, value=111, and jvb saves the parameters mapped by name not allowing null keys). So I'd rather not set it.